web_client/views/VideoWidget.js

```diff
@@ -0,0 +1,92 @@
+import $ from 'jquery';
+import _ from 'underscore';
+import reconcile from 'reconcile.js';
+import View from 'girder/views/View';
+import { getApiRoot, restRequest } from 'girder/rest';
+
+import ModalView from './ModalView';
+
+import template from '../templates/VideoWidget.pug';
+
+import '../stylesheets/VideoWidget.styl';
+
+var VideoWidget = View.extend({
+    events: {
+        'click button.convert': function (e) {
+            this.status = 2;
+            this.playing = false;
+            this.render();
+            setTimeout(() => {
+                this.status = 3;
+                this.render();
+            }, 3000);
+        },
+        'click button.delete': function (e) {
+            this.status = 0;
+            this.playing = false;
+            this.render();
+        },
+        'click button.play': function (e) {
+            this.videoElement.play();
+        },
+        'click button.pause': function (e) {
+            this.videoElement.pause();
+        },
+        'click button.popup': function (e) {
+            new ModalView({
+                el: $('#g-dialog-container'),
+                model: this.fileModel,
+                parentView: this
+            }).render();
+        }
+    },
+    initialize(settings) {
+        this.fileModel = settings.model;
+        this.modalMode = settings.modalMode;
+        this.status = (() => {
+            let status = Math.floor(Math.random() * 3);
```

Comment:
especially here, these code blocks are especially hard to read with these numbers, I have no idea what this means.

Reply:
This feature at this moment is a mock-up, so I just randomly set the initial status of the video processing.